human-tracking-and-pose-detection/helper_for_visualization.py

```diff
@@ -0,0 +1,194 @@
+    'right_elbow': 8,
+    'left_wrist': 9,
+    'right_wrist': 10,
+    'left_hip': 11,
+    'right_hip': 12,
+    'left_knee': 13,
+    'right_knee': 14,
+    'left_ankle': 15,
+    'right_ankle': 16
+}
+
+# Maps bones to a matplotlib color name.
+KEYPOINT_EDGE_INDS_TO_COLOR = {
+    (0, 1): 'm',
+    (0, 2): 'c',
+    (1, 3): 'm',
+    (2, 4): 'c',
+    (0, 5): 'm',
+    (0, 6): 'c',
+    (5, 7): 'm',
+    (7, 9): 'm',
+    (6, 8): 'c',
+    (8, 10): 'c',
+    (5, 6): 'y',
+    (5, 11): 'm',
+    (6, 12): 'c',
+    (11, 12): 'y',
+    (11, 13): 'm',
+    (13, 15): 'm',
+    (12, 14): 'c',
+    (14, 16): 'c'
+}
+
+def _keypoints_and_edges_for_display(keypoints_with_scores,
+                                     height,
+                                     width,
+                                     keypoint_threshold=0.11):
+  """Returns high confidence keypoints and edges for visualization.
+
+  Args:
+    keypoints_with_scores: A numpy array with shape [1, 1, 17, 3] representing
+      the keypoint coordinates and scores returned from the MoveNet model.
+    height: height of the image in pixels.
+    width: width of the image in pixels.
+    keypoint_threshold: minimum confidence score for a keypoint to be
+      visualized.
+
+  Returns:
+    A (keypoints_xy, edges_xy, edge_colors) containing:
+      * the coordinates of all keypoints of all detected entities;
+      * the coordinates of all skeleton edges of all detected entities;
+      * the colors in which the edges should be plotted.
+  """
+  keypoints_all = []
+  keypoint_edges_all = []
+  edge_colors = []
+  num_instances, _, _, _ = keypoints_with_scores.shape
+  for idx in range(num_instances):
+    kpts_x = keypoints_with_scores[0, idx, :, 1]
+    kpts_y = keypoints_with_scores[0, idx, :, 0]
+    kpts_scores = keypoints_with_scores[0, idx, :, 2]
+    kpts_absolute_xy = np.stack(
+        [width * np.array(kpts_x), height * np.array(kpts_y)], axis=-1)
+    kpts_above_thresh_absolute = kpts_absolute_xy[
+        kpts_scores > keypoint_threshold, :]
+    keypoints_all.append(kpts_above_thresh_absolute)
+
+    for edge_pair, color in KEYPOINT_EDGE_INDS_TO_COLOR.items():
+      if (kpts_scores[edge_pair[0]] > keypoint_threshold and
+          kpts_scores[edge_pair[1]] > keypoint_threshold):
+        x_start = kpts_absolute_xy[edge_pair[0], 0]
+        y_start = kpts_absolute_xy[edge_pair[0], 1]
+        x_end = kpts_absolute_xy[edge_pair[1], 0]
+        y_end = kpts_absolute_xy[edge_pair[1], 1]
+        line_seg = np.array([[x_start, y_start], [x_end, y_end]])
+        keypoint_edges_all.append(line_seg)
+        edge_colors.append(color)
+  if keypoints_all:
+    keypoints_xy = np.concatenate(keypoints_all, axis=0)
+  else:
+    keypoints_xy = np.zeros((0, 17, 2))
+
+  if keypoint_edges_all:
+    edges_xy = np.stack(keypoint_edges_all, axis=0)
+  else:
+    edges_xy = np.zeros((0, 2, 2))
+  return keypoints_xy, edges_xy, edge_colors
+
+def draw_prediction_on_image(
```

Comment:
The usage of this function in main.py is commented out. Why? Any unused/commented code should be removed. If this can be used, there should be an option made available to the user with commandline argument.